applications/ConvectionDiffusionApplication/custom_strategies/strategies/explicit_runge_kutta_4_eulerian_convdiff_strategy.h

```diff
@@ -0,0 +1,385 @@
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 1.0;
+        }
+        else if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 2 || r_process_info.GetValue(RUNGE_KUTTA_STEP) == 3) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.5;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Initialize the Runge-Kutta substep.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void InitializeRungeKuttaLastSubStep() override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeRungeKuttaLastSubStep();
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 4) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Finalize the Runge Kutta explicit solver step.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void FinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+        BaseType::FinalizeSolutionStep();
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Execute the OSS step.
+     */
+    virtual void ExecuteOSSStep()
+    {
+        KRATOS_TRY;
+
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = BaseType::pGetExplicitBuilder();
+        const auto& r_lumped_mass_vector = p_explicit_bs->GetLumpedMassMatrixVector();
+
+        // Get model part data
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        const auto n_nodes = r_model_part.NumberOfNodes();
+        const int n_elem = r_model_part.NumberOfElements();
+
+        ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+        auto& r_settings = *p_settings;
+
+        // Initialize the projection value
+#pragma omp parallel for
```

Comment:
Use parallel utils.

Reply:
Done.

Reply:
If you don't mind, please double check such change. It is my first experience with parallel_utils.